### PR TITLE
Blog post for removal of maven.twttr.com

### PR DESCRIPTION
**Pull request by bryce-anderson**

### CODE REVIEW

File: source/blog/2016-11-02-central-libthrift.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Historically Finagle has depended on org.apache.thrift libthrift
+verison 0.5.x, which happens to not be published to the Central
+Repository. We have published the artifact to the maven.twttr.com,
+but this requires users to add the maven.twttr.com repository as a
+resolver for their project. This is normally not a serious problem for
+most users, but historically maven.twttr.com has has a propensity to
+be unreliable and inaccessible from certain locations. Our long term
+goal is to transition to a current version of Apache libthrift which is
+hosted in the Central Repository, but there is a significant amount of
+work we need to do internally because that can be realized.
```
Comment (vkostyukov) [inline source/blog/2016-11-02-central-libthrift.md]: Can we reword the last part here? I'm having troubles reading/understanding this.

File: source/blog/2016-11-02-central-libthrift.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+However, I have good news: Now maven.twttr.com is no longer required! We
```
Comment (vkostyukov) [inline source/blog/2016-11-02-central-libthrift.md]: s/I have/we have?

File: source/blog/2016-11-02-central-libthrift.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: Farewell, maven.twttr.com!
```
Comment (kevinoliver) [inline source/blog/2016-11-02-central-libthrift.md]: <3 <3 <3

File: source/blog/2016-11-02-central-libthrift.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+published: true
+post_author:
+  display_name: Bryce Anderson
+  twitter: banderson
+tags: Finagle, Finatra, Util
+---
+
+Historically Finagle has depended on org.apache.thrift libthrift
```
Comment (kevinoliver) [inline source/blog/2016-11-02-central-libthrift.md]: s/on org.apache.thrift libthrift/on a forked org.apache.thrift libthrift/

File: source/blog/2016-11-02-central-libthrift.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+verison 0.5.x, which happens to not be published to the Central
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/verison/version

File: source/blog/2016-11-02-central-libthrift.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Repository. We have published the artifact to the maven.twttr.com,
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/to the maven.twttr.com/to maven.twttr.com

File: source/blog/2016-11-02-central-libthrift.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+but this requires users to add the maven.twttr.com repository as a
+resolver for their project. This is normally not a serious problem for
+most users, but historically maven.twttr.com has has a propensity to
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/has has/has had

File: source/blog/2016-11-02-central-libthrift.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+be unreliable and inaccessible from certain locations. Our long term
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/long term/longterm

File: source/blog/2016-11-02-central-libthrift.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+goal is to transition to a current version of Apache libthrift which is
+hosted in the Central Repository, but there is a significant amount of
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/in the Central Repository/by Maven Central

File: source/blog/2016-11-02-central-libthrift.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+work we need to do internally because that can be realized.
+
+However, I have good news: Now maven.twttr.com is no longer required! We
+have published a fork of libthrift v0.5.0 to the central repository
+under the 'com.twitter' organization. The class files still reside in
+the 'org.apache' namespace, so users can still upgrade to newer version
+of libthrift as they wish, but doing so will now require manual exclusion
+rules because the eviction mechanisms used by sbt rely on matching
+organization names. See the [sbt documentation on library managment](http://www.scala-sbt.org/0.13/docs/Library-Management.html)
```
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: s/managment/management

Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: do you mind giving an example of how to do this?

File: source/blog/2016-11-02-central-libthrift.md (hunk 1 above)
Comment (bryce-anderson) [inline source/blog/2016-11-02-central-libthrift.md]: Did this help? I think a word or two was misplaced, sorry about that.

File: source/blog/2016-11-02-central-libthrift.md (hunk 9 above)
Comment (bryce-anderson) [inline source/blog/2016-11-02-central-libthrift.md]: I believe org.apache.thrift:libthrift is actually hosted by the Central Repository. Am I mistaken?

File: source/blog/2016-11-02-central-libthrift.md (hunk 10 above)
Comment (bryce-anderson) [inline source/blog/2016-11-02-central-libthrift.md]: As was discussed yesterday in the spring planning meeting, I plan to give an example, but not until I have a way to test it. That means either this blog post waits until we've published artifacts, it gets amended, or there is another smaller one. All things being equal, I'd prefer one of the first two options.

File: source/blog/2016-11-02-central-libthrift.md (hunk 9 above)
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: I don't think there's a difference between "Maven Central" and "Central Repository" but I think it's more commonly referred to as "Maven Central".  I might be wrong on this though.

File: source/blog/2016-11-02-central-libthrift.md (hunk 10 above)
Comment (mosesn) [inline source/blog/2016-11-02-central-libthrift.md]: @bryce-anderson sorry, must have missed that discussion.  you don't think it's sufficient to do a publishLocal and use that artifact?

File: source/blog/2016-11-02-central-libthrift.md (hunk 9 above)
Comment (bryce-anderson) [inline source/blog/2016-11-02-central-libthrift.md]: I've also heard it called both and I don't think there would be any confusion in either case. I'm just trying to stick with the naming convention used at https://search.maven.org/ and http://central.sonatype.org/.

File: source/blog/2016-11-02-central-libthrift.md (hunk 10 above)
Comment (bryce-anderson) [inline source/blog/2016-11-02-central-libthrift.md]: I'm just wary of discrepancies that can happen and don't want to sink a lot of time verifying that there are no shenanigans.

File: source/blog/2016-11-02-central-libthrift.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+hosted by the Central Repository, but there is a significant amount of
+work we need to do internally before we can get there.
+
+However, we have good news: Now maven.twttr.com is no longer required! We
+have published a fork of libthrift v0.5.0 to the central repository
+under the 'com.twitter' organization. The class files still reside in
+the 'org.apache' namespace, so users can still upgrade to newer version
+of libthrift as they wish, but doing so will now require manual exclusion
+rules because the eviction mechanisms used by sbt rely on matching
+organization names. See the [sbt documentation on library management](http://www.scala-sbt.org/0.13/docs/Library-Management.html)
```
Comment (taylorleese) [inline source/blog/2016-11-02-central-libthrift.md]: Can we include example exclusion rules for sbt and maven?

File: source/blog/2016-11-02-central-libthrift.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+Repository. We have published the artifact to maven.twttr.com,
+but this requires users to add the maven.twttr.com repository as a
+resolver for their project. This is normally not a serious problem for
+most users, but historically maven.twttr.com has had a propensity to
+be unreliable and inaccessible from certain locations. Our longterm
+goal is to transition to a current version of Apache libthrift which is
+hosted by the Central Repository, but there is a significant amount of
+work we need to do internally before we can get there.
+
+However, we have good news: Now maven.twttr.com is no longer required! We
```
Comment (taylorleese) [inline source/blog/2016-11-02-central-libthrift.md]: Now => now

File: source/blog/2016-11-02-central-libthrift.md (hunk 11 above)
Comment (taylorleese) [inline source/blog/2016-11-02-central-libthrift.md]: Oh, saw the other thread on this. Ignore.

File: source/blog/2016-11-02-central-libthrift.md (hunk 10 above)
Comment (taylorleese) [inline source/blog/2016-11-02-central-libthrift.md]: I'm fine with waiting until things are published FWIW.

**Comment by bryce-anderson** [comment]
Added an example libraryDependencies config for overriding libthrift.

**Comment by taylorleese** [comment]
🐑 🇮🇹

**Comment by bryce-anderson** [comment]
@taylorleese, I feel pretty sheepish 😉, but I don't know what 🐑 🇮🇹 means.

**Comment by taylorleese** [comment]
@bryce-anderson 🐑 🇮🇹 == 🚢 🇮🇹 == :shipit:

**Pull request merged**